muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/ForumSubscriptionNotifier.java

```diff
@@ -0,0 +1,136 @@
+        Long userId = areaSubscription.getUser();
+        
+        UserEntity recipient = userEntityController.findUserEntityById(userId);
+
+        ForumThreadSubscription forumThreadSubscription = forumThreadSubsciptionController.findByThreadAndUserEntity(forumThread, recipient);
+        
+        if (forumThreadSubscription != null) {
+          continue;
+        }
+        
+        if ((poster != null) && (recipient != null)) {
+          if (!Objects.equals(poster.getId(), recipient.getId())) {
+            Map<String, Object> params = new HashMap<String, Object>();
+            User senderUser = userController.findUserByUserEntityDefaults(poster);
+            params.put("sender", String.format("%s", senderUser.getDisplayName()));
+            params.put("subject", forumThread.getTitle());
+            params.put("content", forumThread.getMessage());
+            if (event.getWorkspaceUrlName() != null) {
+              params.put("url", String.format("%s/workspace/%s/discussions#" + forumArea.getId(), event.getBaseUrl(), event.getWorkspaceUrlName()));
+
+            } else {
+              params.put("url", String.format("%s/discussion#" + forumArea.getId(), event.getBaseUrl()));
+            }
+            notifierController.sendNotification(forumThreadNewMessageNotification, poster, recipient, params);
+          }
+      }
+      }
+    }
+    
+    List<ForumThreadSubscription> threadSubscriptions = forumThreadSubsciptionController.listByThread(forumThread);
+    
+    if (threadSubscriptions != null) {
+      for (ForumThreadSubscription threadSubscription : threadSubscriptions) {
+        Long userId = threadSubscription.getUser();
+        
+        UserEntity recipient = userEntityController.findUserEntityById(userId);
+
+        if ((forumThread != null) && (poster != null) && (recipient != null)) {
+          if (!Objects.equals(poster.getId(), recipient.getId())) {
+            Map<String, Object> params = new HashMap<String, Object>();
+            User senderUser = userController.findUserByUserEntityDefaults(poster);
+            params.put("sender", String.format("%s", senderUser.getDisplayName()));
+            params.put("subject", forumThread.getTitle());
+            params.put("content", forumThread.getMessage());
+            
+            if (event.getWorkspaceUrlName() != null) {
+              params.put("url", String.format("%s/workspace/%s/discussions#" + forumArea.getId() + "/1/" + forumThread.getId(), event.getBaseUrl(), event.getWorkspaceUrlName()));
+
+            } else {
+              params.put("url", String.format("%s/discussion#" + forumArea.getId() + "/1/" + forumThread.getId(), event.getBaseUrl()));
+            }
+            notifierController.sendNotification(forumThreadNewMessageNotification, poster, recipient, params);
+          }
+      }
+    }
+    
+    } else {
+      logger.log(Level.SEVERE, String.format("Forum couldn't send notifications as some entity was not found"));
```

Comment:
Indentation seems a little off. Also, isn't "some entity" in this case just `threadSubscriptions` being null? Sounds like an unlikely scenario as wouldn't the method simply return an empty list if the thread has no subscriptions?